x-pack/metricbeat/module/gcp/metrics/metrics_requester_test.go

```diff
@@ -85,7 +85,7 @@ func TestGetFilterForMetric(t *testing.T) {
 			"storage",
 			"storage.googleapis.com/firewall/dropped_bytes_count",
 			metricsRequester{config: config{Region: "us-central1"}, logger: logger},
-			"metric.type=\"storage.googleapis.com/firewall/dropped_bytes_count\" AND resource.labels.location = \"us-central1\"",
+			"metric.type=\"storage.googleapis.com/firewall/dropped_bytes_count\" AND resource.label.location = starts_with(\"us-central1\")",
```

Comment:
can we increase coverage in order to include all the scenarios that the refactored method can produce?